common/presenters/population-table/population-table-config.js

```diff
@@ -0,0 +1,73 @@
+const { addDays, format, isSameDay } = require('date-fns')
+
+const locationMetaToSpacesLink = require('./location-meta-to-spaces-link')
+
+const populationTableDay = {
+  establishmentConfig({ isSortable = false }) {
+    return {
+      head: {
+        isSortable,
+        sortKey: 'title',
+        html: 'Establishment',
+        attributes: {
+          width: '220',
+        },
+      },
+      row: {
+        attributes: {
+          scope: 'row',
+        },
+        html: data => {
+          return `<a>${data.title}</a>`
+        },
+      },
+    }
+  },
+
+  dayConfig({ focusDate, baseDate, baseOffset = 0 }) {
+    const dateWithOffset = addDays(baseDate, baseOffset)
+
+    return {
+      head: populationTableDay.dayHeadConfig({
+        date: dateWithOffset,
+        focusDate,
+      }),
+      row: populationTableDay.dayRowConfig({
+        date: dateWithOffset,
+        focusDate,
+        populationIndex: baseOffset,
+      }),
+    }
+  },
+
+  dayHeadConfig({ date, focusDate }) {
+    return {
+      text: format(date, 'EEE dd'),
+      attributes: {
+        width: '80',
+      },
+      classes: isSameDay(focusDate, date)
+        ? 'focus-table__tr--day focus-table__tr--focus'
+        : 'focus-table__tr--day',
+    }
+  },
+
+  dayRowConfig({ date, focusDate, populationIndex = 0 }) {
+    const spacesHrefPrefix = '/'
+
+    return {
+      html: data => {
+        return locationMetaToSpacesLink({
```

Comment:
Given `locationMetaToSpacesLink` is now a lot do you think it's still worth extracting it?

This could simplify this function and reduce the amount of jumping around needed to understand what's going on. 

```javascript
dayRowConfig({ date, focusDate, populationIndex = 0 }) {
  return {
    html: data => {
      const count = data.meta.populations[populationIndex].free_spaces
      const action = count === undefined 
        ? i18n.t('population::add_space') 
        : i18n.t('population::spaces_with_count', { count })
      
      return `<a href="/">${action}</a>`
    },
    classes: isSameDay(focusDate, date)
      ? 'focus-table__td--day focus-table__td--focus'
      : 'focus-table__td--day',
  }
})
```

**Disclaimer:** I don't have the full context so you may have extracted this to separate function as it will be re-used elsewhere.

Reply:
It was separate for testing reasons, but you're right, I think it's been refactored away anyway.

I've done a variant of what you've got above, but with a thought to the future work involving links, i.e. I don't think a one liner will end up being as readable once that's done.